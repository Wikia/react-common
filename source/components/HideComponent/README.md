Hide:
```js
<HideComponent isHidden>Default</HideComponent>
```

Display
```js
<HideComponent>Default</HideComponent>
```

