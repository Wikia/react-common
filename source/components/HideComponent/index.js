import PropTypes from 'prop-types';

const HideComponent = ({ hide, children }) => {
    if (hide === true) {
        return false;
    }

    return children;
}

HideComponent.propTypes = {
    hide: PropTypes.bool,
};

HideComponent.defaultProps = {
    hide: false,
};

export default HideComponent;
