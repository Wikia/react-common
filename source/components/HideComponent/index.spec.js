import React from 'react';
import { mount } from 'enzyme';

import HideComponent from './index';

test('Hide component tree', () => {
    const component = mount(
        <HideComponent hide>
            <div> test </div>
        </HideComponent>,
    );
    expect(component).toMatchSnapshot();
});

test('Show component tree', () => {
    const component = mount(
        <HideComponent>
            <div> test </div>
        </HideComponent>,
    );
    expect(component).toMatchSnapshot();
});
