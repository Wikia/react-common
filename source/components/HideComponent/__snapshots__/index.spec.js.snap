// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hide component tree 1`] = `
<HideComponent
  hide={true}
/>
`;

exports[`Show component tree 1`] = `
<HideComponent
  hide={false}
>
  <div>
     test 
  </div>
</HideComponent>
`;
